src/librustdoc/passes/calculate_doc_coverage.rs

```diff
@@ -41,10 +41,13 @@ impl ItemCount {
         has_docs: bool,
         has_doc_example: bool,
         should_have_doc_examples: bool,
+        should_have_docs: bool,
     ) {
-        self.total += 1;
+        if should_have_docs {
+            self.total += 1;
+        }
 
-        if has_docs {
+        if has_docs && should_have_docs {
```

Comment:
Same reasoning here.
```suggestion
        if has_docs {
```